pokemongo_bot/__init__.py

```diff
@@ -121,55 +121,63 @@ def _setup_api(self):
         print('[#] UltraBalls: ' + str(balls_stock[3]))
         self.get_player_info()
 
-        if self.config.firsttrans:
-            self.first_transfer()
-            
+        if self.config.initial_transfer:
+            self.initial_transfer()
+
         print('[#]')
         self.update_inventory();
-        
-    def first_transfer(self):
-        print('[x] First Transfer.')
-        
-        pokemon_groups = self._first_transfer_get_groups()
-        
-        print('[x] Transfering...')
-        
+
+    def initial_transfer(self):
+        print('[x] Initial Transfer.')
+
+        if self.config.cp:
+            print('[x] Will NOT transfer anything above CP {}'.format(self.config.cp))
+        else:
+            print('[x] Preparing to transfer all Pokemon duplicates, keeping the highest CP of each one type.')
+
+        pokemon_groups = self._initial_transfer_get_groups()
+
         for id in pokemon_groups:
-        
+
             group_cp = pokemon_groups[id].keys()
-            
             if len(group_cp) > 1:
                 group_cp.sort()
                 group_cp.reverse()
 
                 for x in range(1, len(group_cp)):
+                    if self.config.cp and group_cp[x] > self.config.cp:
+                        continue
+
+                    print('[x] Releasing Pokemon #{} and CP {}'.format(id, group_cp[x]))
                     self.api.release_pokemon(pokemon_id=pokemon_groups[id][group_cp[x]])
                     response_dict = self.api.call()
-                    time.sleep(2)
-                    
+                    sleep(2)
+
         print('[x] Transfering Done.')
-        
-    def _first_transfer_get_groups(self):
+
+    def _initial_transfer_get_groups(self):
         pokemon_groups = {}
         self.api.get_player().get_inventory()
         inventory_req = self.api.call()
-        inventory_dict = inventory_req['responses']['GET_INVENTORY']['inventory_delta']['inventory_items']  
-        
+        inventory_dict = inventory_req['responses']['GET_INVENTORY']['inventory_delta']['inventory_items']
+
         for pokemon in inventory_dict:
             try:
```

Comment:
Whole block is:
```
try:
     reduce(dict.__getitem__, ["inventory_item_data", "pokemon_data", "pokemon_id"], pokemon)
except KeyError:
      continue
```

Which is basically a test to see if key at `pokemon['inventory_item_data']['pokemon_data']['pokemon_id']` can be pulled correctly. If it cannot, it means this is not a pokemon (probably an egg), so we skip it.